webapp/components/notifications/NotificationMenu/NotificationMenu.vue

```diff
@@ -29,13 +41,37 @@ export default {
   },
   data() {
     return {
+      displayedNotifications: [],
       notifications: [],
     }
   },
   props: {
     placement: { type: String },
   },
+  watch: {
+    notifications: {
+      immediate: true,
+      handler(lastQueriedNotifications) {
+        if (lastQueriedNotifications && lastQueriedNotifications.length > 0) {
+          lastQueriedNotifications.forEach(updatedListNotification => {
+            const sameNotification = this.displayedNotifications.find(oldListNotification => {
+              return this.equalNotification(oldListNotification, updatedListNotification)
+            })
+            if (!sameNotification) this.displayedNotifications.unshift(updatedListNotification)
```

Comment:
Instead of `unshift` I would use `concat` and then `sort` by `createdAt`.

https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/concat
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort

Who knows if the notifications arrive in the right order?

Reply:
when you say who knows, can you explain that? 
we are currently asking for them back with `orderBy: updatedAt`
doesn't that mean either we do know the order they arrive or something is broken in the functionality of either `neo4j-graphql-js` and/or `neo4j`?

Reply:
Sorting is not necessary in my eyes.
Should be the right way anyhow and if not in my eyes its better to have the new notifications at the beginning.

Reply:
I don't know exactly if it could happen in this very example. But since we merge the `read` notifications on the client which are not part of the response that comes from the server (thus not guaranteed to be sorted) I find it good practice to order the list on the client again.

Here's a (very speculative) real world example. Consider we are connected to the fediverse. A notification was created for the user on the other server.

1. Sb. mentions you on a remote server
2. You fetch the notifications
3. The notifications from the remote server arrive at your server
4. You fetch the notifications again. Now the old notification gets put on top even though the `createdAt` is older. This would break the order of the list by `createdAt`.

Reply:
You don't *have* to implement it. It was an intuition of mine, because we merge something sorted into something not sorted. Feel free to ignore this suggestion

Reply:
I like your explanation, I hadn't thought of that. I think it doesn't hurt to do it either, I'm sure it's not expensive, especially since the list is more or less sorted anyways

Reply:
I sorted now!